src/shared/mkfs-util.c

```diff
@@ -620,7 +620,8 @@ int make_filesystem(
 
         } else if (streq(fstype, "erofs")) {
                 argv = strv_new(mkfs,
-                                "-U", vol_id);
+                                "-U", vol_id,
+                                "--hard-dereference");
```

Comment:
The option seems something new.
If erofs-utils is old, I guess mkfs will fail...